src/query.jl

```diff
@@ -80,12 +80,18 @@ function query(filename; checkfile::Bool=true)
         elseif !checkfile && lensym(sym) > 1
             return File{DataFormat{sym[1]}}(filename)
         end
-        if no_magic && !hasfunction(sym)
+        no_function = !hasfunction(sym)
+        if no_magic && no_function
             error("Some formats with extension ", ext, " have no magic bytes; use `File{format\"FMT\"}(filename)` to resolve the ambiguity.")
         end
+        if no_magic && !no_function
+            # try specific function first, if available
+            ret = query(open(filename), abspath(filename), sym)
+            ret != nothing && return file!(ret)
```

Comment:
```suggestion
            ret !== nothing && return file!(ret)
```
This ends up calling `===` which is a builtin and hence doesn't use runtime dispatch even in cases like this one where inferrability is nonexistent. For a singleton like `nothing` this is safe.

Reply:
I repeatedly make this mistake.. thanks

Reply:
No worries, and the landscape has already changes since I wrote that: with https://github.com/JuliaLang/julia/pull/38905 merged, starting in Julia 1.7 `isnothing` and `ismissing` could once again become reasonable choices!

Reply:
Also it's not really a mistake to use `!=`, it's just a missed opportunity for better inferrability.